### DISCUSSION TITLE
Cannot sync. Error : src refspec main does not match any...

### DISCUSSION THREAD

**Discussion by Su24nn3** [trilbymedia/grav-plugin-git-sync#208] [Q&A] [answered]
Hi ! 

I am getting "src refspec main does not match any..." after trying to sync to a repo on the same server. 
I am using Gitlab, I have allowed localhost webhooks access. 

Any idea what might be wrong ? 

If you need me to provide more info I'll be happy to.

Thanks !

**Comment by w00fz** [comment, marked as answer]
This could be caused by various reasons. Try checking the suggestions found [here](https://www.akashmittal.com/error-src-refspec-master-not-match/)

**Reply by Su24nn3** [reply]
Thanks ! 
I tried `git show-ref`, and noticed several refs, with both `refs/remotes/origin/main` and `refs/remotes/origin/master`. This lead me to realise my local branch was actually named master, whereas in the grav admin panel for GitSync it was set as "main" (I set it this way for the remote during configuration, because my Gitlab creates main branch instead of master by default, but I don't know why it did the same for local branch). Do you think this might be an issue, or did I miss something during configuration ?

**Reply by w00fz** [reply]
You can rename your local branch by simply running the command `git branch -m master main` - this should be sufficient since your remote origin branch is already named `master`.

One other thing I started doing was to change my local global setting for the default branch to `main` because, as you already found out, it does default to `master`.

You can do the same by running the command: `git config --global init.defaultBranch main`

**Comment by RadTechDad** [comment]
I'm getting the same error. When i'm in the webroot, and I run `git show-ref` I get the following error:
`fatal: Not a git repository (or any of the parent directories): .git`

Am I supposed to do a `git init` first? Shouldn't the plugin handle that since it's keeping in sync with a remote server?

**Reply by w00fz** [reply]
You are supposed to run the command in the user folder, gitsync operates in there

**Reply by RadTechDad** [reply]
Awesome! So I went into the `user` folder and ran `git show-ref` and saw that the main branch name was called `master`:
`refs/heads/master`

So what you mentioned before works:
1. I renamed the branch from `master` to `main` by using the `git branch -m master main` command.
2. Then I made sure that all other branches will use the `main` branch in the future by running:
`git config --global init.defaultBranch main`

Thank you!
